### PR TITLE
expand_dims function

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented expand_dims function.

### CODE REVIEW

File: chainer/functions/array/expand_dims.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        self.axis = int(axis)
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type, = in_types
+        if self.axis >= 0:
+            type_check.expect(x_type.ndim >= self.axis)
+        else:
+            type_check.expect(x_type.ndim >= -self.axis - 1)
+
```
Comment (delta2323) [inline chainer/functions/array/expand_dims.py]: Both NumPy and CuPy's `expand_dims` can accept `axis` that does not satisfy `-ndim-1 <= axis <= ndim`. If either of inequalities does not hold, the value of `axis` is regarded as an edge value.

**Comment by delta2323** [comment]
Could you consider the case I commented?

**Comment by unnonouno** [comment]
Negative axis behaves strangely. So, I made the condition.

```
>>> x=numpy.zeros((2,3,4))
>>> x.shape
(2, 3, 4)
```

Positive case:
```
>>> numpy.expand_dims(x, 0).shape
(1, 2, 3, 4)
>>> numpy.expand_dims(x, 1).shape
(2, 1, 3, 4)
>>> numpy.expand_dims(x, 2).shape
(2, 3, 1, 4)
>>> numpy.expand_dims(x, 3).shape
(2, 3, 4, 1)
>>> numpy.expand_dims(x, 4).shape
(2, 3, 4, 1)
```

And negative case:
```
>>> numpy.expand_dims(x, -1).shape
(2, 3, 4, 1)
>>> numpy.expand_dims(x, -2).shape
(2, 3, 1, 4)
>>> numpy.expand_dims(x, -3).shape
(2, 1, 3, 4)
>>> numpy.expand_dims(x, -4).shape
(1, 2, 3, 4)
>>> numpy.expand_dims(x, -5).shape
(2, 3, 1, 4)  # ???
>>> numpy.expand_dims(x, -6).shape
(2, 1, 3, 4)  # ???
>>> numpy.expand_dims(x, -7).shape
(1, 2, 3, 4)  # ???
>>> numpy.expand_dims(x, -8).shape
(1, 2, 3, 4)
>>> numpy.expand_dims(x, -9).shape
(1, 2, 3, 4)
```

**Comment by delta2323** [comment]
@unnonouno  I remembered you reported this behavior in the offline discussion. Then, it is acceptable to restrict the range of axis until it is needed.

**Comment by delta2323** [comment]
@unnonouno  Could you add the test cases for invalid boundary condition on axes?

**Comment by unnonouno** [comment]
OK, I added a test case.

**Comment by delta2323** [comment]
LGTM :+1:
